### ISSUE TITLE
Semantics of p_c_pointer and PointerToCPtr

### ISSUE THREAD

**Issue by certik** [lcompilers/lpython#592]
Currently `p_c_pointer(pointer(xi32), p)` gets translated to:
```
         (= 
            (Var 6 p) 
            (PointerToCPtr 
               (GetPointer 
                  (Var 6 xi32) 
                  (Pointer 
                     (Integer 4 [])) ()) 
               (CPtr) ()) ()) 
```
and `p_c_pointer(xi64, p)` gets translated to:
```
         (= 
            (Var 6 p) 
            (PointerToCPtr 
               (Var 6 xi64) 
               (CPtr) ()) ()) 
```
What should be the recommended usage? It seems to me the second case is incorrect, as `PointerToCPtr` should only accepts pointers as the first argument?

The other issue is: we need to also have a CPython implementation of whatever syntax/semantics we choose. If it cannot be done for the above syntax, then we need to change it.

**Comment by czgdp1807** [comment]
We can use `ctypes` to run the code with `CPython`?

**Comment by certik** [comment]
Yes, we already do in `ltypes.py`.

**Comment by czgdp1807** [comment]
I see. So just importing stuff from `ctypes` into `ltypes` will allow CPython to at least run the code. However, I am wondering can CPython really call into C code. Like AFAIK, LPython generates an object file and that's linked to the object file of `.c` file (where we implement the C functions called in Python code). Now since CPython never generates an object file then how will the C functions linked to their definition.

**Comment by certik** [comment]
Yes, so callbacks cannot be done from CPython, I also just realized that, so we will simply not implement `@ccallable` in CPython. But calling into C is possible, and it works already. So all other features we'll get working.
